akka-actor-tests/src/test/scala/akka/util/VersionSpec.scala

```diff
@@ -32,6 +32,9 @@ class VersionSpec extends AnyWordSpec with Matchers {
       new Version("1.2.3-foo") should !==(new Version("1.2.3-bar"))
       new Version("1.2-foo") should be > new Version("1.2.3")
       new Version("1.2.3-foo") should be > new Version("1.2.3-bar")
+
+      // dynver format
+      new Version("1.0.10+3-1234abcd") should be < new Version("1.0.11")
```

Comment:
Patrik, I don't mean to be _that guy_ but the following case is still failing:

```
      new Version("1.0.10+2-1234abcd") should be < new Version("1.0.10+10-1234abcd")
```

The parser limits to only three the segments interpreted as numbers.

Reply:
ok, we should implement some smarter parsing and support the dynver format
but let's decide if we want this at all first

Reply:
and corresponding maven plugin, and maybe [our recommendation for the version.number](https://doc.akka.io/docs/akka-management/current/kubernetes-deployment/building-using-maven.html) should be more like dynver format. I think it has the needed properties https://github.com/git-commit-id/git-commit-id-maven-plugin/blob/master/maven/docs/using-the-plugin-in-more-depth.md

Reply:
fixed